### PR TITLE
feat: addPoliciesWithAffected & removePoliciesWithAffected

### PR DESCRIPTION
**Pull request by Shivansh-yadav13**
- ref: #9 
Signed-off-by: Shivansh-yadav13 <yadavshivansh@gmail.com>

### CODE REVIEW

File: src/model/model.ts [hunk 1]
```diff
@@ -445,4 +445,32 @@ export class Model {
...
+    }
+    return effected;
+  }
+
+  // removePoliciesWithAffected removes policies and returns only the affected policies
+  public removePoliciesWithAffected(sec: string, ptype: string, rules: string[][]): string[][] {
+    const effected: string[][] = [];
+    for (const rule of rules) {
+      const ok = this.hasPolicy(sec, ptype, rule);
+      if (!ok) {
```
Comment (nodece) [inline src/model/model.ts:468]:
```suggestion
      if (ok) {
```

File: test/managementAPI.test.ts [hunk 2]
```diff
@@ -162,6 +162,24 @@ test('addPolicies', async () => {
...
 });
 
+test('addPoliciesWithAffected', async () => {
+  const a = getStringAdapter('examples/rbac_policy.csv');
+  e.setAdapter(a);
+  const rules = [
+    ['jack', 'data4', 'read'],
+    ['katy', 'data4', 'write'],
+    ['leyo', 'data4', 'read'],
+    ['ham', 'data4', 'write'],
```
Comment (nodece) [inline test/managementAPI.test.ts]: I think you should add a duplicate rule to test to avoid adding duplicate rule.

File: test/managementAPI.test.ts [hunk 3]
```diff
@@ -232,6 +250,26 @@ test('removePolicies', async () => {
...
+  ];
+  const added = await e.addPolicies(rules);
+  expect(added).toBe(true);
+  const removed = await e.removePoliciesWithAffected(rules);
```
Comment (nodece) [inline test/managementAPI.test.ts:273]: Add a rule that does not exist in the model to the rules, then assert.

**Review by nodece — APPROVED** [review]
